Ensure Metadata is Properly Passed to LogReturn in Logging Class

imabutahersiddik opened the pull request:
This commit addresses an issue in the logging class where the metadata parameter was not consistently passed to the LogReturn instance within the _log method.

Resolves #46

Keyrxng commented on src/logs.ts:
```diff
@@ -12,14 +12,15 @@ export class Logs {
         level,
       },
-      metadata
+      metadata // Ensure metadata is passed correctly here
```
metadata was already passed in here, you've just added a comment

Keyrxng commented:
@imabutahersiddik Hello bud, this feature should fill the `revision` property of the `Metadata` object and should pull it from cloudflare as mentioned in the spec. For actions it can be pulled easily from the most up to date sha in relation to what branch the action ran on

imabutahersiddik commented:
Solved: Metadata is Properly Passed to LogReturn in Logging

Changes Made: Added a new trace log level in logs.ts, implemented the corresponding trace method in PrettyLogs, and replaced the trace symbol with "🔍".

0x4007 reviewed (commented):
There's nothing about trace in the spec. @gentlementlegen rfc
